### PR TITLE
Complete Project

### PR DESCRIPTION
**Pull request by kendoriddy**
**In this milestone, I achieved the following...**

A quick description of the project can be found [here](https://www.loom.com/share/c75b20a437174962be40057fd011c2a9)

- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (catagories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### CODE REVIEW

File: app/views/categories/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      </ul>
+ 
+   
+   <% else %>
+          <p>No Category Added yet</p> 
+      <% end %>
+  <div>
+ <div class='category-footer'>
+    <%= button_to "Add Category", new_category_path, method: :get, class: 'category-btn' %>
+  </div>
```
Comment (Deepakdanger) [inline app/views/categories/index.html.erb:48]: - [ ] I am unable to see the button text.

Screenshot from 2022-12-24 01-37-20

File: app/views/categories/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        <div class="transaction-container">
+    <hr>
+    <% @transactions.each do |transaction| %>
+    <div class='transaction'>
+      <span>
+        <h3><%= transaction.name %></h3>
+        <small>$<%= transaction.amount %></small>
+        <small><%= transaction.created_at.to_date.strftime("%d %b, %Y") %></small>
+      </span>
+      <span>
```
Comment (Deepakdanger) [inline app/views/categories/show.html.erb]: - [ ] The design and font are not looking as per the design. Please fix that.

### your design
Screenshot from 2022-12-24 01-32-40

### given design

Screenshot from 2022-12-24 00-59-59

File: app/views/users/registrations/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,50 @@
+<section class="user-input-container">
+   <header class="input-container">
+    <nav>
+      <ul class='links'>
+        <li>
+        <p><%= link_to "←", splash_screens_index_path, class:'text-light text-decoration-none' %></p>
+        <li>
+        <li>
```
Comment (Deepakdanger) [inline app/views/users/registrations/new.html.erb:8]: - [ ] I can't able to see the text. Please fix the UI text color.

Screenshot from 2022-12-24 01-36-04

File: db/migrate/20221222100021_change_user_id_to_author_id_in_categories.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class ChangeUserIdToAuthorIdInCategories < ActiveRecord::Migration[7.0]
+  def change
+    change_table :categories do |t|
+      # t.index :author_id
+      # t.references :author, null: false, foreign_key: { to_table: :users }
```
Comment (Deepakdanger) [inline db/migrate/20221222100021_change_user_id_to_author_id_in_categories.rb]: - [ ] I would recommend removing the commented code to make the codebase cleaner.

File: db/migrate/20221222102027_remove_user_id_from_categories.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+class RemoveUserIdFromCategories < ActiveRecord::Migration[7.0]
+  def change
+    change_table :categories do |t|
+      # t.remove :user_id
+    end
```
Comment (Deepakdanger) [inline db/migrate/20221222102027_remove_user_id_from_categories.rb]: - [ ] I would recommend removing the commented code to make the codebase cleaner.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @kendoriddy, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks good :+1: 

## Required changes
_Check the comments under the review._


## Optional changes
You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app/views/categories/index.html.erb (hunk 1 above)
Comment (kendoriddy) [reply app/views/categories/index.html.erb:48]: Thanks for your review @Deepakdanger I will work on them all

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Kenny, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- linters are passing

 ## Required Changes ♻️
_Check the comments under the review._
- [ ] because the previous CR didn't notice I'll review the project as usual however it seems you don't have any linters set-up for your project, please make sure to do so before requesting another review.
- [ ] the live version of your project is not working as expected please make sure it works.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kendoriddy** [comment]
Screen Shot 2022-12-23 at 22 27 24
Thanks for your review @mikethreels. I have linters set up for the project, both stylint and rubocop.  Attached is a screenshot  of a linters check for the project. And as for the live version, I am not able to deploy to Render as it keeps asking for my card details and rejecting it when I input it.

**Pull request merged**